### ISSUE TITLE
API: Problem with slashes in identifiers yet again, but this time for the `/product-models` GET API

### ISSUE THREAD

**Issue by ishasaran** [akeneo/pim-community-dev#9141]
Hi there,

This is a recurring problem from the fix that was deployed which has the introduction of the `requirement` as an option to specify a regex for allowing a certain character(s), in the identifiers of the api. The identifiers has slashes ("/").

**Expected behaviour**
To allow the slashes ("/") when hitting the GET API for `/product-models/{code}`, where the `code` identifier can be allowed any character(s) using `requirements` option for `code`, in the YML as,
```
pim_api_product_model_get:
  path: /product-models/{code}
    defaults: { _controller: pim_api.controller.product_model:getAction, _format: json }
    methods: [GET]
    requirements:
        code: .+
```

**The problem**
As mentioned as the solution (https://bit.ly/2qYaLH7) to the issue #6264, we tried adding the `requirement` to our api,
`/api/rest/v1/product-models/{our-product-model-code}`

but we encounter the following error as response,
```
{
    "code": 404,
    "message": "No route found for \"GET /api/rest/v1/product-models/{our-product-model-code}\""
}
```

**Steps to reproduce**
Please be advised to test with the GET a product model API for the PIM (https://api.akeneo.com/api-reference-22.html#get_product_models__code_)

**Comment by ahocquard** [comment]
Hello,

Thanks for raising this issue.
I will talk with our Product Owner in order to know if allowing slash is intentional or not.

Could you provide me your use-case about using a slash in the product model code?

Also, did you clear your cache after modification?

Regards,
Alexandre

**Comment by ishasaran** [comment]
@ahocquard 
Hi and thank you for responding on our query.

The use-case if you're asking is just like how it was a problem for Product's identifier were having slashes and when using them as `code` in the API didn't allow them as a character.
`PIM-7394: add slash as an allowed character for the identifier of a product in the API`
Reference: https://bit.ly/2qYaLH7 (as mentioned in the issue)

Similarly, one or more Product Model's identifier have slashes ("/") and need to be passed as `code` to use the Product Model's API, as currently it isn't identified as a character.

Also, yes we did clear the cache after modification.

**Comment by Serfenia** [comment]
@ahocquard What is the status on this? We're experiencing the same behaviour on the `/api/rest/v1/products/{code}` endpoint.

**Comment by ahocquard** [comment]
@Serfenia sorry, I was quite busy. It has not been fixed for now.

But you should not have this problem on product endpoint, as raised in this issue: https://github.com/akeneo/pim-community-dev/issues/6264

**Comment by Serfenia** [comment]
@ishasaran This issue is still an issue to us. This is a problem as we have to change our main identifier completely or have to contact our supplier to talk about a new SKU pattern for those products.

**Comment by Doodoune** [comment]
Hi @ishasaran & @Serfenia 

Sorry for the delay, this behavior has been corrected in v2.3.34 and pull-up in v3.0.9. :tada: 
=> cf https://github.com/akeneo/pim-community-dev/pull/9761 && https://github.com/akeneo/pim-community-dev/pull/9766

I close this issue. If you still have a problem, don't hesitate to reopen this ticket.

Regards,
